dali/operators/reader/nvdecoder/CMakeLists.txt

```diff
@@ -16,3 +16,51 @@
 collect_headers(DALI_INST_HDRS PARENT_SCOPE)
 collect_sources(DALI_OPERATOR_SRCS PARENT_SCOPE)
 collect_test_sources(DALI_OPERATOR_TEST_SRCS PARENT_SCOPE)
+
+DETERMINE_GCC_SYSTEM_INCLUDE_DIRS("c++" "${CMAKE_CXX_COMPILER}" "${CMAKE_CXX_FLAGS}" INFERED_COMPILER_INCLUDE)
+
+# transform a list of paths into a list of include directives
+set(DEFAULT_COMPILER_INCLUDE)
+foreach(incl_dir ${INFERED_COMPILER_INCLUDE})
+  set(DEFAULT_COMPILER_INCLUDE "${DEFAULT_COMPILER_INCLUDE} -I${incl_dir}")
+endforeach(incl_dir)
+separate_arguments(DEFAULT_COMPILER_INCLUDE UNIX_COMMAND  "${DEFAULT_COMPILER_INCLUDE}")
+
+set(NVCUVID_GENERATED_STUB "${CMAKE_CURRENT_BINARY_DIR}/dynlink_nvcuvid_gen.cc")
+add_custom_command(
+    OUTPUT ${NVCUVID_GENERATED_STUB}
+    COMMAND python ${CMAKE_SOURCE_DIR}/tools/stub_generator/stub_codegen.py --unique_prefix=Nvcuvid --
+                "${CMAKE_SOURCE_DIR}/tools/stub_generator/nvcuvid.json" ${NVCUVID_GENERATED_STUB}
+                "${CMAKE_CURRENT_SOURCE_DIR}/nvcuvid.h" "-I${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
+                "-I${CMAKE_SOURCE_DIR}/include" "-I${CMAKE_SOURCE_DIR}"
+                ${DEFAULT_COMPILER_INCLUDE}
+    DEPENDS ${CMAKE_SOURCE_DIR}/tools/stub_generator/stub_codegen.py
+            "${CMAKE_CURRENT_SOURCE_DIR}/nvcuvid.h"
+            "${CMAKE_CURRENT_SOURCE_DIR}/cuviddec.h"
+            "${CMAKE_SOURCE_DIR}/tools/stub_generator/nvcuvid.json"
+    COMMENT "Running nvcuvid.h stub generator"
+    VERBATIM)
+
+set_source_files_properties(${NVCUVID_GENERATED_STUB} PROPERTIES GENERATED TRUE)
+
+# there is a one cuvidGetVideoSourceState function that doesn't follow the returned value convention
+# so need a dedicated stub for it
+set(NVCUVID_2_GENERATED_STUB "${CMAKE_CURRENT_BINARY_DIR}/dynlink_nvcuvid_2_gen.cc")
```

Comment:
why do we have a `_2` stub?

Reply:
We have a one method that doesn't follow the common and we need a separate json for it.

Reply:
Out wrappers returns CUresult and CUDA_ERROR_SHARED_OBJECT_SYMBOL_NOT_FOUND is symbol is not available.
`cuvidGetVideoSourceState` returns cudaVideoState so we need to return cudaVideoState_Error in case of lack of the symbol.

Reply:
Can't we add one more level of JSON that would aggregate functions, return type and error code? This doesn't seem too complicated - we'd have a dictionary that maps a function to a proper return type and error code and we'd generate stubs with these. I think we might run into this issue in the future.

Reply:
Done